[WebGPU] empty shader source passes WGSL::staticCheck (

mwyrzykowski opened the pull request:
#### 7aec01bbe63727c65ca6b059fda17f0844681f08
<pre>
[WebGPU] empty shader source passes WGSL::staticCheck (
<a href="https://bugs.webkit.org/show_bug.cgi?id=265924">https://bugs.webkit.org/show_bug.cgi?id=265924</a>
&lt;radar://119230505&gt;

Reviewed by NOBODY (OOPS!).

The CTS sends empty source programs to createShaderModule.

A ShaderModule requires a valid entryPoint and a ShaderModule
with empty source string can not have one by definition, so
create an invalid ShaderModule when an empty string is passed
to createShaderModule.

* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::Device::createShaderModule):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7aec01bbe63727c65ca6b059fda17f0844681f08

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29304 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7974 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30639 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31878 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26630 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10114 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5273 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26629 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29577 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6627 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25084 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5705 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5844 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33218 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26756 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26565 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32063 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5812 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3991 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29847 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7501 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6328 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6336 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

tadeuzagallo commented on Source/WebGPU/WGSL/WGSL.cpp:
```diff
@@ -68,6 +68,9 @@ namespace WGSL {
 {
     PhaseTimes phaseTimes;
+    if (!wgsl.length())
+        return FailedCheck { };
```
I believe it is valid to have an empty program: https://www.w3.org/TR/WGSL/#recursive-descent-syntax-translation_unit

mwyrzykowski replied:
Sounds good, I moved the validation instead to `createShaderModule`, since a ShaderModule without an entryPoint is invalid.

mwyrzykowski committed:
[WebGPU] empty shader source passes WGSL::staticCheck (
https://bugs.webkit.org/show_bug.cgi?id=265924
<radar://119230505>

Reviewed by NOBODY (OOPS!).

The CTS sends empty source programs to createShaderModule.

A ShaderModule requires a valid entryPoint and a ShaderModule
with empty source string can not have one by definition, so
create an invalid ShaderModule when an empty string is passed
to createShaderModule.

* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::Device::createShaderModule):
